demo/src/pages/channels/channelProvider.tsx

```diff
@@ -0,0 +1,200 @@
+        childWindows[windowName].close();
+        delete childWindows[windowName];
+      }
+      return state;
+    case "setWindow": {
+      return {
+        ...state,
+        childWindows: {
+          [action.payload.identity.name]: action.payload,
+          ...state.childWindows,
+        },
+      };
+    }
+    case "onConnection": {
+      const { statuses } = state;
+      const { identity } = action.payload;
+      return {
+        ...state,
+        statuses: statuses.concat({
+          msg: `Client connected: ${JSON.stringify(identity)}`,
+          name: identity.name,
+        }),
+      };
+    }
+    case "onDisconnection": {
+      const { statuses } = state;
+      const { identity } = action.payload;
+      return {
+        ...state,
+        statuses: statuses.filter((x: { name: any }) => x.name !== identity.name),
+      };
+    }
+    case "setPushMessage": {
+      return {
+        ...state,
+        pushMessage: action.payload,
+      };
+    }
+    default:
+      return state;
+  }
+};
+
+const ChannelProvider: React.FC = () => {
+  const [state, dispatch] = useReducer(reducer, {
+    childWindows: {},
+    count: 0,
+    statuses: [],
+  });
+  const channelActions = [
+    {
+      action: () => dispatch({ type: "increment" }),
+      topic: "increment",
+    },
+    {
+      action: () => dispatch({ type: "decrement" }),
+      topic: "decrement",
+    },
+    {
+      action: (payload: any) => dispatch({ type: "close", payload }),
+      topic: "close",
+    },
+  ];
+
+  const { provider } = useChannelProvider(CHANNEL_NAME, channelActions);
+  useEffect(() => {
+    if (provider) {
+      provider.onConnection((identity: any) => {
+        dispatch({ type: "onConnection", payload: { identity } });
+      });
+      provider.onDisconnection((identity: any) => {
+        dispatch({ type: "onDisconnection", payload: { identity } });
+      });
+    }
+    Prism.highlightAll();
+  }, [provider]);
+
+  const createWindow = async () => {
+    const newWindow = await window.fin.Window.create({
+      autoShow: true,
+      defaultHeight: 420,
+      defaultTop: 50 + Object.keys(state.childWindows).length * 20,
+      defaultWidth: 680,
+      frame: false,
+      name: uuidv4(),
+      url: "/demo/channel-client",
+    });
+
+    dispatch({ type: "setWindow", payload: newWindow });
+  };
+
+  const { statuses, pushMessage, childWindows } = state;
+  return (
+    <div className={styles.container}>
+      <h1 className={styles.title}>useChannelProvider</h1>
+      <p>
+        This hook creates a ChannelProvider for the given channelName and
+        registers any actions supplied to it. The actions can then be dispatched
+        from the connected ChannelClient using the useChannelClient hook.
+      </p>
+      <div className={styles.description}>
+        <i>
+          This hook automatically unsubscribes whenever your component unmounts.
+        </i>
+      </div>
+      <h2>Try it out</h2>
```

Comment:
All other hook demos have the "Try it out" section after the code example section, could we keep it consistent?